## Final Project Submission

Please fill out:
* Student names: Christopher Noel, Margaret Nyairo, Victor Masinde, James Ngumo, Anthony Ekeno. 
* Student pace: full time 
* Instructor name: Maryann Mwikali


 Market Analysis & Insights For Strategic Movie Production

## Business Understanding

### Overview
This project is designed to aid a company's venture into the movie production industry by launching a new studio. Through comprehensive data analysis, the project will identify current trends and provide actionable insights from box office data. This information will guide the company in determining the types of movies that are most successful in today’s market, thereby supporting strategic content creation and maximizing box office returns

###  The Problem Statement
The company needs to pinpoint what types of movies are most successful in the current market to propel the new studio's launch. Specifically, we aim to:

1. **Identify the Genres Performing Well at the Box Office**: Determine which movie genres are currently popular and yield high box office returns.
   
2. **Analyze Movie Budgets and Profitability**: Evaluate the relationship between production budgets, returns, and overall profitability to find the optimal investment range.
   
3. **Assess Audience Demographics Driving Success**: Understand which demographic segments are contributing significantly to box office revenues.
   
4. **Recommend Optimal Release Seasons or Windows**: Identify the best times of the year for releasing movies to maximize box office performance.

### Main objectives
To identify the most successful types of films currently at the box office and translate these insights into actionable strategies that guide the new movie studio's production choices, ensuring competitive and commercial success in the film industry.


 ## Data Wrangling
 


 ### import libraries
First we import the necessary packages for Exploratory data analysis.


In [1]:
# Importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3 

%matplotlib inline


### Exploratory Data Analysis

The data used in this analysis contains data collected from various popular movie sites such as Box Office Mojo, IMDb, rotten tomato reviews. It contains detailed information on movie titles, actors, directors, box office earnings, and movie ratings.
1. im.db

2. box office mojo

3. movie budgets 

4. movie info 


 

### Import Data sets



##### First Dataset - im.db
This dataset will form root basis of analysis

In [2]:
#FIRST DATA SET IM.DB- 

#establishing a connection with database.
conn = sqlite3.connect("zippedData/im.db")
cur = conn.cursor()

#opening the database

pd.read_sql("""
SELECT*
FROM sqlite_master
""",conn) 

type           name       tbl_name  rootpage  \
0  table   movie_basics   movie_basics         2   
1  table      directors      directors         3   
2  table      known_for      known_for         4   
3  table     movie_akas     movie_akas         5   
4  table  movie_ratings  movie_ratings         6   
5  table        persons        persons         7   
6  table     principals     principals         8   
7  table        writers        writers         9   

                                                 sql  
0  CREATE TABLE "movie_basics" (\n"movie_id" TEXT...  
1  CREATE TABLE "directors" (\n"movie_id" TEXT,\n...  
2  CREATE TABLE "known_for" (\n"person_id" TEXT,\...  
3  CREATE TABLE "movie_akas" (\n"movie_id" TEXT,\...  
4  CREATE TABLE "movie_ratings" (\n"movie_id" TEX...  
5  CREATE TABLE "persons" (\n"person_id" TEXT,\n ...  
6  CREATE TABLE "principals" (\n"movie_id" TEXT,\...  
7  CREATE TABLE "writers" (\n"movie_id" TEXT,\n  ...

###### An Entity Relationship Diagram [ERD]
-Below is an ERD explaining further contents contained in tables shown above

-After studying each column name in the ERD below, we discover that we are interested in the contenct of two tables. These are; **movie_basics and movie_ratings**  since they have contents that will be vital to our analysis.

-Analyse structure of the tables by using **JOIN** statement to combine them.

![Alt text](movie_data_erd.jpeg)

######  JOIN tables[ movie basics + movie ratings]  to create a new dataframe  **imdb**

In [3]:
## merge required tables then convert it to a dataframe
#Select relevant information from movie_basics table
#JOIN to movie_ratings

imdb = pd.read_sql("""
SELECT primary_title,start_year,runtime_minutes,genres,averagerating
FROM movie_basics
JOIN movie_ratings
USING("movie_id")
""",conn)
imdb

primary_title  start_year  runtime_minutes  \
0                            Sunghursh        2013            175.0   
1      One Day Before the Rainy Season        2019            114.0   
2           The Other Side of the Wind        2018            122.0   
3                      Sabse Bada Sukh        2018              NaN   
4             The Wandering Soap Opera        2017             80.0   
...                                ...         ...              ...   
73851                 Diabolik sono io        2019             75.0   
73852                Sokagin Çocuklari        2019             98.0   
73853                        Albatross        2017              NaN   
73854       La vida sense la Sara Amat        2019              NaN   
73855                       Drømmeland        2019             72.0   

                     genres  averagerating  
0        Action,Crime,Drama            7.0  
1           Biography,Drama            7.2  
2                     Drama            6.9  
3              Comedy,Drama            6.1  
4      Comedy,Drama,Fantasy            6.5  
...                     ...            ...  
73851           Documentary            6.2  
73852          Drama,Family            8.7  
73853           Documentary            8.5  
73854                  None            6.6  
73855           Documentary            6.5  

[73856 rows x 5 columns]

##### SECOND DATASET  -box office mojo gross
Essential for understanding box office success and profitability.

In [4]:
## SECOND DATASET 
gross = pd.read_csv("zippedData/bom.movie_gross.csv.gz")
gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

##### THIRD DATASET -movie budgets
Used to assess budget-related profitability.

In [5]:
## THIRD DATASET
budget = pd.read_csv("zippedData/tn.movie_budgets.csv.gz")
budget.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

###### FOURTH DATASET - movie info
Allows us to categorize movies, essential for identifying high-performing genres


In [6]:
# FOURTH DATASET
movie_info = pd.read_csv("zippedData/rt.movie_info.tsv.gz",sep="\t")
movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

####  Data Understanding 
In this section we are going to examine our data for better understanding before we start working on it. The section helps us ascertain the number of rows and columns(`.shape`). We also get a slight summary of the data displaying column names, number of non-null values and the Dtype of the column contents(`info`).When dealing with continous data, we can have a brief statistical summary of the columns with intergers or float data types(`.describe`)

######  .info Function
This returns the summary of the data frame

In [7]:
# first dataset
imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   primary_title    73856 non-null  object 
 1   start_year       73856 non-null  int64  
 2   runtime_minutes  66236 non-null  float64
 3   genres           73052 non-null  object 
 4   averagerating    73856 non-null  float64
dtypes: float64(2), int64(1), object(2)
memory usage: 2.8+ MB


In [8]:
# second dataset 
gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [9]:
#  third dataset
budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [10]:
# fourth dataset 
movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


###### .columns Function
This returns column labels of each dataframe

In [11]:
# First DataSet
imdb.columns

Index(['primary_title', 'start_year', 'runtime_minutes', 'genres',
       'averagerating'],
      dtype='object')

In [12]:
# Second DataSet
gross.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [13]:
# Third DataSet
budget.columns

Index(['id', 'release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [14]:
# Fourth DataSet 
movie_info.columns

Index(['id', 'synopsis', 'rating', 'genre', 'director', 'writer',
       'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime',
       'studio'],
      dtype='object')

###### .describe Function 
This returns the descriptive statistics of the dataframe

In [15]:
# First DataSet
imdb.describe()

start_year  runtime_minutes  averagerating
count  73856.000000     66236.000000   73856.000000
mean    2014.276132        94.654040       6.332729
std        2.614807       208.574111       1.474978
min     2010.000000         3.000000       1.000000
25%     2012.000000        81.000000       5.500000
50%     2014.000000        91.000000       6.500000
75%     2016.000000       104.000000       7.400000
max     2019.000000     51420.000000      10.000000

In [16]:
# Second DataSet 
gross.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

In [17]:
# Third Dataset
budget.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

In [18]:
# Fourth DataSet
movie_info.describe()

id
count  1560.000000
mean   1007.303846
std     579.164527
min       1.000000
25%     504.750000
50%    1007.500000
75%    1503.250000
max    2000.000000

###### .shape  function 
-This returns number of rows , columns

In [19]:
# First DataSet
imdb.shape

(73856, 5)

In [20]:
# Seond DataSet
gross.shape

(3387, 5)

In [21]:
# Third DataSet
budget.shape

(5782, 6)

In [22]:
# Fourth DataSet
movie_info.shape

(1560, 12)

We are working with four different datasets for this project. The first one named **imdb** has 73856 rows and 5 columns.It was sourced from internet. It has all the three types of data namely Float, intergers and objects. The data here helps us analyse the `genres, titles, runtime minutes and the ratings of movies`

The second dataset is named **gross** and it has 3387 rows and 5 columns. It was sourced from internet. It has Float, intergers and objects as dataypes. The data here helps us analyse income generated as it contains columns with `domestic and foreign gross data.`

The third dataset is named **budget** and it has 5782 rows and 6 columns. It was sourced from internet. The datatypes in this dataset are intergers and objects only. It has columns that can help us calculate the budget of producing a movie i.e `production_budget`. We can also see seasonal trends as it has a column with information on dates. 

The last dataset is named **movie_info** and it has 1560 rows and 12 columns. It is sourced from internet. The datatypes in this dataset are intergers and objects only. It has columns with information about `writer, director, studio` etc that can help us make informed reccomendations at the end of the project


# Data cleaning
 Modify the data set in some manner to correct erroneous data, remove redundancies;
 
 PART A

 checking for `missing values`
 
 checking for `Duplicates`
 
 PART B
 
 deciding whether to fill/drop missing values& duplicates
 
 dropping Columns {based(1&2)/ on irrelevance }

#### Part A 
##### checking for missing values in DataSets

In [23]:
## First DataSet %missing_Values
imdb.isnull().mean()*100

primary_title       0.000000
start_year          0.000000
runtime_minutes    10.317374
genres              1.088605
averagerating       0.000000
dtype: float64

In [24]:
# Second DataSet %missing_Values
gross.isnull().mean()*100

title              0.000000
studio             0.147623
domestic_gross     0.826690
foreign_gross     39.858282
year               0.000000
dtype: float64

In [25]:
# Third DataSet %missing_Values
budget.isnull().mean()*100

id                   0.0
release_date         0.0
movie                0.0
production_budget    0.0
domestic_gross       0.0
worldwide_gross      0.0
dtype: float64

In [26]:
# Fourth DataSet %missing_Values
movie_info.isnull().mean()*100

id               0.000000
synopsis         3.974359
rating           0.192308
genre            0.512821
director        12.756410
writer          28.782051
theater_date    23.012821
dvd_date        23.012821
currency        78.205128
box_office      78.205128
runtime          1.923077
studio          68.333333
dtype: float64

####  check for duplicates in DataSets

In [27]:
# FirstDataSet -duplicates--has 1 duplicate
imdb.duplicated().value_counts()

False    73855
True         1
dtype: int64

In [28]:
# SecondDataSet - duplicates--has no duplicates
gross.duplicated().value_counts()

False    3387
dtype: int64

In [29]:
# ThirdDataSet -duplicates--has no duplicates
budget.duplicated().value_counts()

False    5782
dtype: int64

In [30]:
# FourthDataSet -duplicates--has no duplicates
movie_info.duplicated().value_counts()

False    1560
dtype: int64

### Part B  
-Drop `duplicates`

-Drop `rows` with missing_values

-Drop `columns` based on relevance & consistency

In [31]:
# code drops duplicated rows in the imdb DataSet
imdb.drop_duplicates(inplace=True)
imdb.duplicated().value_counts()

False    73855
dtype: int64

In [32]:
# # The code below drop rows with missing values permanently
imdb.dropna(subset=['runtime_minutes',  'genres',], inplace=True)
imdb

primary_title  start_year  runtime_minutes  \
0                            Sunghursh        2013            175.0   
1      One Day Before the Rainy Season        2019            114.0   
2           The Other Side of the Wind        2018            122.0   
4             The Wandering Soap Opera        2017             80.0   
6                      Joe Finds Grace        2017             83.0   
...                                ...         ...              ...   
73849        Padmavyuhathile Abhimanyu        2019            130.0   
73850                     Swarm Season        2019             86.0   
73851                 Diabolik sono io        2019             75.0   
73852                Sokagin Çocuklari        2019             98.0   
73855                       Drømmeland        2019             72.0   

                           genres  averagerating  
0              Action,Crime,Drama            7.0  
1                 Biography,Drama            7.2  
2                           Drama            6.9  
4            Comedy,Drama,Fantasy            6.5  
6      Adventure,Animation,Comedy            8.1  
...                           ...            ...  
73849                       Drama            8.4  
73850                 Documentary            6.2  
73851                 Documentary            6.2  
73852                Drama,Family            8.7  
73855                 Documentary            6.5  

[65719 rows x 5 columns]

In [33]:
# Check to confirm no more missing values in dataframe
imdb.isna().sum()

primary_title      0
start_year         0
runtime_minutes    0
genres             0
averagerating      0
dtype: int64

##### Dropping Columns
-`movie_info & budget` , have a common column with the same name `id`

-On further analysis the columns have no corelation. To retain consistency we drop 

In [34]:
## drop 'id' column form budget DataSet
budget.drop(columns=['id'], inplace=True)

###### For movie_info we decide to drop the columns  currency & box office , since they had 78% missing values and would cause inconsistencies in our data

In [35]:
#We drop columns that have been noted 
movie_info.drop(columns=['currency', 'box_office', 'id'], inplace=True)

### Data Transformation
-Converting the datatypes in our datasets to enable manipulation

-`gross[foreign_gross]-&-budget[domestic_gross,producion_budget,worldwide_gross]`  the columns in this dataset are  `object` datatype,this means the numeric figures are read like `strings` instead of numbers. So to have clarity ,we convert them to `float datatype` for more accurate readings

In [36]:
# Second DataSet-- gross
# Changing data type from an object to a float

gross['foreign_gross'] = gross['foreign_gross'].replace({'\$': '', ',': ''}, regex=True).astype(float)
gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   float64
 4   year            3387 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 132.4+ KB


###### converting budget columns`domestic_gross`, `production_budget`, `worldwide_gross` to `float datatype`


In [37]:
# List of columns to clean and convert to float
columns_converted = ['domestic_gross', 'production_budget', 'worldwide_gross']

# Loop through each column, replace the unwanted characters, and convert to float
for column in columns_converted:
    budget[column] = budget[column].replace({'\$': '', ',': ''}, regex=True).astype(float)

In [38]:
budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   release_date       5782 non-null   object 
 1   movie              5782 non-null   object 
 2   production_budget  5782 non-null   float64
 3   domestic_gross     5782 non-null   float64
 4   worldwide_gross    5782 non-null   float64
dtypes: float64(3), object(2)
memory usage: 226.0+ KB
